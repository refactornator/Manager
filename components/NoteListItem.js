import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import timeago from 'timeago.js';

export default ({ note }) => (
  <View style={styles.item}>
    <Text style={styles.timestamp}>{timeago().format(note.createdAt)}</Text>
    <Text style={styles.text}>{note.text}</Text>
  </View>
);

const styles = StyleSheet.create({
  item: {
    flex: 1,
    padding: 2,
    marginTop: 2,
    borderColor: 'grey',
    borderWidth: 2
  },
  timestamp: {
    marginBottom: 2
  },
  text: {
    fontSize: 20
  }
});
