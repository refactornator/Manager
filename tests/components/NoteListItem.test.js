import React from 'react';

import renderer from 'react-test-renderer';

import NoteListItem from '../../components/NoteListItem';

const note = {
  key: 'abc',
  text: 'This is my note content.',
  createdAt: new Date('05/22/1987')
};

it('renders a note list item', () => {
  const rendered = renderer
    .create(<NoteListItem note={note} deleteHandler={() => {}} />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

describe('when the close button is clicked', () => {
  it('calls the delete handler passing the note key as a parameter', () => {});
});
