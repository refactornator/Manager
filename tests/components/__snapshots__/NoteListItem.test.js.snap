// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a note list item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "flex": 1,
      "marginBottom": 6,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 6,
      "paddingBottom": 6,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 6,
      "shadowColor": "#0E2451",
      "shadowOffset": Object {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 0.2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 10,
            "marginBottom": 4,
          },
          undefined,
        ]
      }
    >
      30 years ago
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#CED1D6",
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ÔÄç
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 16,
          "marginBottom": 4,
        },
        undefined,
      ]
    }
  >
    This is my note content.
  </Text>
</View>
`;
